lib/rouge/lexers/turing.rb

```diff
@@ -0,0 +1,164 @@
+        Brush Button CheckBox Comm Concurrency
+        Config Dir Draw DropBox EditBox
+        Error ErrorNum Event File Font
+        GUI Input Joytick Keyboard Limits
+        ListBox Math Menu Mouse Music
+        Net Obsolete PC Pen Pic
+        Print RadioButton Rand RGB Sound
+        Sprite Str Stream Student Sys
+        Text Time TypeConv Video View
+        Window
+      )
+
+      constants = %w(
+        black blue brightblue brightcyan brightgreen
+        brightmagenta brightpurple brightred brightwhite brown
+        brushErrorBase cdMaxNumColors cdMaxNumColours cdMaxNumPages cdScreenHeight
+        cdScreenWidth clLanguageVersion clMaxNumDirStreams clMaxNumRunTimeArgs clMaxNumStreams
+        clRelease cmFPU cmOS cmProcessor colorbg
+        colourbg colorfg colourfg configErrorBase cyan
+        darkgray darkgrey defFontID defWinID dirErrorBase drawErrorBase
+        errWinID fileErrorBase fontDefaultID
+        fontErrorBase fsysErrorBase generalErrorBase gray green
+        grey guiErrorBase joystick1 joystick2 lexErrorBase
+        magenta mouseErrorBase musicErrorBase penErrorBase
+        placeCenterDisplay placeCentreWindow printerErrorBase purple
+        red rgbErrorBase spriteErrorBase streamErrorBase textErrorBase
+        timeErrorBase unixSignalToException viewErrorBase white windowErrorBase
+        yellow true false
+      )
+
+      reserved = %w(
+        addressint all asm
+        assert begin bind bits body
+        break by case
+        cheat checked class close collection
+        condition const decreasing def deferred
+        else elseif elsif end
+        endfor endif endloop exit
+        export external fcn
+        for fork forward free function
+        get handler if implement
+        inherit init
+        invariant label
+        loop module monitor
+        new
+        objectclass of opaque open
+        packed pause pervasive pointer post
+        pre priority proc procedure process
+        put quit read
+        record register result
+        return seek self set
+        signal skip tag
+        tell then timeout to
+        unchecked union unqualified
+        wait when write
+      )
+
+      import = %w(import include)
+
+      declarations = %w(
+        var enum type unit
+      )
+
+      types = %w(
+        boolean char string
+        real real4 real8
+        int int1 int2 int4
+        nat nat1 nat2 nat4
+        flexible array
+      )
+
+      operators = %w(
+        \+ - \* \/
+        < > = :
+        \^ #
+        ~
+        & \|
+        \\( \\)
+        , \.
+      )
+
+      wordOperators = %w(
+        div mod rem
+        not and or xor
+        in
+        shl shr
+      )
+
+      state :root do
+        rule /[^\S\n]+/, Text
+	rule /\n/, Text
+
+        rule /#{operators.join('|')}/, Operator
+
+        rule %r(%.*?$), Comment::Single
+        rule %r(/\*.*?\*/)m, Comment::Multiline
+
+        rule /(?:#{constants.join('|')})\b/, Keyword::Constant
```

Comment:
Please see how this is done in other lexers, with a Set constructed separately, combined with a single rule and set lookups. This is done because Rouge matches each regex in turn, so reducing the number of regex matches is preferable so that a string can be excluded quickly. E.g. search for "rule id" in the C lexer.